src/fluree/db/api/branch.cljc

```diff
@@ -0,0 +1,196 @@
+            [fluree.db.util.branch :as util.branch]
+            [fluree.db.util.ledger :as util.ledger]
+            [fluree.db.util.log :as log]))
+
+(defn create-branch!
+  "Creates a new branch from an existing branch.
+  
+  Parameters:
+    conn - Connection object
+    new-branch-spec - Full branch spec (e.g., 'ledger:new-branch')
+    from-branch-spec - Source branch spec (e.g., 'ledger:old-branch')
+    from-commit - (optional) Specific commit id (sha256 URI) to branch from, defaults to latest
+    
+  Returns the new branch metadata."
+  [conn new-branch-spec from-branch-spec from-commit]
+  (go-try
+    (let [[ledger-id new-branch] (util.ledger/ledger-parts new-branch-spec)
+          [from-ledger-id from-branch] (util.ledger/ledger-parts from-branch-spec)]
+
+      (when (not= ledger-id from-ledger-id)
+        (throw (ex-info "Cannot create branch across different ledgers"
+                        {:status 400 :error :db/invalid-branch-operation})))
+
+      ;; Load source ledger to get its current commit
+      (let [source-ledger (<? (connection/load-ledger conn from-branch-spec))
+            source-db (ledger/current-db source-ledger)
+            ;; Prefer commit ID (sha256 URI) for lineage as it's consistent across storage backends
+            source-commit-id (or from-commit (get-in source-db [:commit :id]))
+
+            ;; Create branch metadata
+            metadata {:created-at (util/current-time-iso)
+                      :source-branch from-branch
+                      :source-commit source-commit-id}
+
+            ;; Prepare commit for new branch with flat metadata fields
+            source-commit-map (:commit source-db)
+            compact-commit (-> source-commit-map
+                               commit-data/->json-ld
+                               (assoc "alias" new-branch-spec
+                                      "branch" new-branch)
+                               (util.branch/augment-commit-with-metadata metadata))
+
+            primary-publisher (:primary-publisher conn)
+            secondary-publishers (:secondary-publishers conn)
+            _ (log/debug "create-branch! publishing commit for" new-branch-spec
+                         "with primary-publisher?" (boolean primary-publisher))
+            _ (when primary-publisher
+                (log/debug "Publishing commit with alias:" (get compact-commit "alias")
+                           "address:" (get compact-commit "address")
+                           "t:" (get-in compact-commit ["data" "t"]))
+                (<? (nameservice/publish primary-publisher compact-commit))
+                (log/debug "Published commit for" new-branch-spec)
+                ;; Also publish to secondary publishers asynchronously
+                (nameservice/publish-to-all compact-commit secondary-publishers))]
+
+        (util.branch/branch-creation-response new-branch metadata source-commit-id)))))
+
+(defn- same-ledger?
```

Comment:
This terminology is confusing to me. I had been using `ledger-alias` as the full name including the branch, and `ledger-name` as the name without the branch, and `ledger` as the catch all for all of a ledgers branches treated as a single unit.

The name of this function and the intermediate bindings used within it do not follow that model. I'm fine with changing my internal terminology, but I do think we should pick consistent terms for those concepts and stick to them.